include/hip/hcc_detail/hip_cooperative_groups_helper.h

```diff
@@ -106,21 +106,21 @@ __CG_STATIC_QUALIFIER__ uint32_t size() {
 }
 
 __CG_STATIC_QUALIFIER__ uint32_t thread_rank() {
-  // Compute global id of the workgroup to which the current threads belongs to
+  // Compute global id of the workgroup to which the current thread belongs to
   uint32_t blkIdx =
            (uint32_t)((hipBlockIdx_z * hipGridDim_y * hipGridDim_x) +
-                      (hipBlockIdx_y * hipGridDim_x) +
+                      (hipBlockIdx_x * hipGridDim_y) +
```

Comment:
This does not seem right. I believe the previous version was correct. Could you describe why this change would be correct?

And *if* this change is correct (which I don't think it is), then why not change the line above as well? Why remove hipBlockIdx_y from the calculation, but leave hipBlockIdx_z?

Reply:
Actually I got confused :) Let me think again, and correct it in case, if I made mistake.